CRM/Utils/System/Joomla.php

```diff
@@ -210,7 +212,8 @@ public function setTitle($title, $pageTitle = NULL) {
     $template = CRM_Core_Smarty::singleton();
     $template->assign('pageTitle', $pageTitle);
 
-    $document = JFactory::getDocument();
+    $factoryClassName = $this->factoryClassName();
+    $document = $factoryClassName::getDocument();
```

Comment:
`Factory::getDocument()` is deprecated in Joomla 4.3. Suggest replacing with Factory::getApplication()->getDocument().

https://api.joomla.org/cms-4/classes/Joomla-CMS-Factory.html#method_getDocument

(This occurs in several places, I won't comment on each one!)